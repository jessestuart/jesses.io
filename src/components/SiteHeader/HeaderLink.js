/* @flow */
import { Link } from 'gatsby'
import React from 'react'

import classNames from 'classnames'

type Props = {
  children: *,
  className?: *,
  href: *,
  isActive: *,
}

const HeaderLink = ({ children, className, href, isActive = false }: Props) => {
  const headerClassNames: string = classNames(
    'lato f4 ph3 mv2 pv2 flex-auto',
    className,
    {
      'white-90 fw4': isActive,
      'light-silver fw3': !isActive,
    }
  )
  return (
    <ul className={headerClassNames}>
      <Link className="shadow-none flex justify-center items-center" to={href}>
        {children}
      </Link>
    </ul>
  )
}

export default HeaderLink
