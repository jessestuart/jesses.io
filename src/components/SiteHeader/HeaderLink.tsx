import classNames from 'classnames'
import Link from 'gatsby-link'
import _ from 'lodash'
import React, { ReactNode } from 'react'
import { Text } from 'rebass'
import styled from 'styled-components'
import { ThemeProvider } from 'theme-ui'

import { HeaderTheme } from 'components/SiteHeader/SiteHeader'
import StyledLink from 'components/StyledLink/StyledLink'
import Theme from 'styles/Theme'

const StyledHeaderLink = styled(Link)`
  transition: all 0.5s;
  &:hover {
    color: ${Theme.colors.textLight};
  }
`

interface Props {
  children?: ReactNode
  className?: string
  href: string
  isActive?: boolean
  pathname?: string
}

const getIsActive = ({ href, pathname }): boolean =>
  (href === '/' && pathname === '/') ||
  (href !== '/' && _.startsWith(pathname, href))

const getThemeForPathname = (pathname: string): HeaderTheme =>
  _.startsWith(pathname, '/photography') ? HeaderTheme.LIGHT : HeaderTheme.DARK

const HeaderLink = (props: Props) => {
  const { children, className, href, pathname } = props
  const isActive = getIsActive({ href, pathname })
  const theme = getThemeForPathname(pathname)

  return (
    <StyledHeaderLink
      className={classNames(
        'b--mid-gray f4 fw3 mv2 ph3 pv2 shadow-none',
        {
          'black-90 fw5': isActive && theme === HeaderTheme.LIGHT,
          'light-silver': !isActive && theme === HeaderTheme.LIGHT,
          'light-silver fw3': !isActive && theme === HeaderTheme.DARK,
          'white-90 fw4': isActive && theme === HeaderTheme.DARK,
        },
        className,
      )}
      to={href}
    >
      {children}
    </StyledHeaderLink>
  )
}

export default HeaderLink
