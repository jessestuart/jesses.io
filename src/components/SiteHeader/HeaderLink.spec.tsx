import React from 'react'

import renderer from 'react-test-renderer'

import HeaderLink from './HeaderLink'

jest.mock('react-headroom', () => 'Headroom')

test('Render HeaderLink component w/ default values.', () => {
  const component = renderer.create(
    <HeaderLink href="/" pathname="/">
      Header Link
    </HeaderLink>,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
