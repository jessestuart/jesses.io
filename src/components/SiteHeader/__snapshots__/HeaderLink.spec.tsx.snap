// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render HeaderLink component w/ default values. 1`] = `
.c0 {
  -webkit-transition: color 0.3s;
  transition: color 0.3s;
}

<a
  className="c0 f4 fw3 mv2 ph3 pv2 shadow-none white-90 fw4 b--mid-gray"
  href="/true"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Header Link
</a>
`;
